luatest/server.lua

```diff
@@ -226,9 +227,12 @@ function Server:build_env()
     local res = {
         TARANTOOL_WORKDIR = self.workdir,
         TARANTOOL_HTTP_PORT = self.http_port,
-        TARANTOOL_LISTEN = self.net_box_port or self.net_box_uri,
         TARANTOOL_ALIAS = self.alias,
     }
+    local listen = self.net_box_port or self.net_box_uri
+    if listen ~= nil then
+        res.TARANTOOL_LISTEN = json.encode(listen)
+    end
```

Comment:
Here we break compatibility with commands other than the default (server_instance.lua).

Reply:
True. Now table is converted to string using `uri.format(uri.parse(table), true)`.